src/core/execute.h

```diff
@@ -91,6 +91,7 @@ typedef enum ExecKeyringMode {
 struct ExecStatus {
         dual_timestamp start_timestamp;
         dual_timestamp exit_timestamp;
+        dual_timestamp handover_timestamp;
```

Comment:
so this doesn't really belong here, because ExecStatus collects information about all individual processes we fork off as part of unit management. But this field here only exists for the main process, no?

i find this really weird to mangle this into the generic ExecStatus structure (whose only purpose is to be reused for each type of process we execute) if we don't actually fill it in except for one specific case.

Hence, either drop this from here, and move it into the generic unit code as a singleton.

Or alternatively, make it work so that it actually works for everything single process we fork off, including ExecStartPre=, ExecStartPost=, ExecStop= and so on, like the start/exit timestamps do.

But that would probably mean using a different fd than the exec_fd one, since that one really only makes sense for executing the main process, as it is a synchronization concept for just that. 

(and there is only a single one of those exec_fds, and we'd need multiple since ExecReload/ExecStop= can run in parallel with ExecStart=...

Reply:
It's quite obvious why in this struct: because that's what `main_exec_status` in the service struct uses. But I don't mind where it lives of course, so I'll just move it to the parent object.